database_cleanup/tests/test_database_cleanup.py

```diff
@@ -78,3 +89,9 @@ def test_database_cleanup(self):
         with self.assertRaises(ProgrammingError):
             with self.registry.cursor() as cr:
                 self.env.cr.execute('select * from database_cleanup_test')
+        with self.registry.cursor() as cr2:
+            # Release the delete of ir_module_module pending
+            self.env.cr.rollback()
+            cr2.execute(
+                "DELETE FROM ir_module_module WHERE id=%s", (module.id,))
```

Comment:
I see. Well, it's not beautiful, but I don't think it's worth the time to get this cursor's commit patched away to by starting a patch fest of the involved methods. Ignore this comment then